zulipterminal/model.py

```diff
@@ -1591,7 +1593,31 @@ def _handle_update_message_event(self, event: Event) -> None:
         indexed_message = self.index["messages"].get(message_id, None)
 
         if indexed_message:
-            self.index["edited_messages"].add(message_id)
+            if "orig_content" in event:
+                self.index["edited_messages"].add(message_id)
+            if "prev_stream" in event:
+                self.index["moved_messages"].add(message_id)
+            if "subject" in event:
+                if not event["subject"].startswith(RESOLVED_TOPIC_PREFIX):
```

Comment:
This looks very similar to the logic in the previous commit. Once you've simplified both, are they simple enough to combine into one function?